File: exercises/exercise-1-testing-java/src/main/java/io/miragon/example/application/port/outbound/newsletter/NewsletterSubscriptionRepository.java
package io.miragon.example.application.port.outbound.newsletter;

import io.miragon.example.domain.NewsletterSubscription;
import io.miragon.example.domain.SubscriptionId;

public interface NewsletterSubscriptionRepository {
    NewsletterSubscription find(SubscriptionId subscriptionId);
    NewsletterSubscription search(SubscriptionId subscriptionId);
    void save(NewsletterSubscription subscription);
    void delete(SubscriptionId subscriptionId);
}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/io/miragon/example/application/port/outbound/newsletter/NewsletterSubscriptionRepository.kt
package io.miragon.example.application.port.outbound.newsletter

import io.miragon.example.domain.newsletter.NewsletterSubscription
import io.miragon.example.domain.SubscriptionId

interface NewsletterSubscriptionRepository {
    fun find(subscriptionId: SubscriptionId): NewsletterSubscription
    fun search(subscriptionId: SubscriptionId): NewsletterSubscription?
    fun save(subscription: NewsletterSubscription)
    fun delete(subscriptionId: SubscriptionId)
}